### PR TITLE
Serialize, deserialize and persist frame tracks between captures

### PR DESCRIPTION
**Pull request by ronaldfw**
Frame tracks are now serialized when saving captures and are properly
deserialized when loading a capture. The user choice of a frame track
is also persisted between captures, that is, when a user selects a
function as a frame track, and takes a new capture, the frame track
is properly added at the end of the capture (what is still missing
are live updates to the frame track during capture).

Bug: http://b/171026228
Tested: Unit tests; manual tests (capture/save/load, etc.)

### CODE REVIEW

File: OrbitCaptureClient/CaptureClient.cpp [hunk 1]
```diff
@@ -39,7 +39,8 @@ ErrorMessageOr<void> CaptureClient::StartCapture(
     ThreadPool* thread_pool, const ProcessData& process,
```
Comment (dimitry-) [inline OrbitCaptureClient/CaptureClient.cpp:39]: Bug: http://b/...

**Review by florian-kuebler — APPROVED** [review]
Makes a lot sense. LGTM! Thanks!

File: OrbitClientModel/CaptureDeserializer.cpp [hunk 2]
```diff
@@ -126,6 +126,7 @@ void LoadCaptureInfo(const CaptureInfo& capture_info, CaptureListener* capture_l
   for (const auto& module : capture_info.modules()) {
     orbit_grpc_protos::ModuleInfo module_info;
     module_info.set_name(module.name());
+    LOG("module added: %s", module.file_path());
```
[the remark this answers is not in the document]
Comment (ronaldfw) [reply OrbitClientModel/CaptureDeserializer.cpp:129]: No, thanks for spotting this. Left over from debugging.l

File: OrbitCore/UserDefinedCaptureData.h [hunk 3]
```diff
@@ -12,11 +12,13 @@
...
 class UserDefinedCaptureData {
  public:
-  [[nodiscard]] FunctionInfoSet frame_track_functions() const { return frame_track_functions_; }
+  [[nodiscard]] const FunctionInfoSet& frame_track_functions() const {
+    return frame_track_functions_; }
   void InsertFrameTrack(const orbit_client_protos::FunctionInfo& function);
   void EraseFrameTrack(const orbit_client_protos::FunctionInfo& function);
   [[nodiscard]] bool ContainsFrameTrack(const orbit_client_protos::FunctionInfo& function) const;
-  void Clear() { frame_track_functions_.clear(); }
+  void Clear() {
```
[the remark this answers is not in the document]
Comment (ronaldfw) [reply OrbitCore/UserDefinedCaptureData.h:20]: Thanks, fixed!

**Review by dimitry- — APPROVED** [review]
otherwise lgtm

File: OrbitCaptureClient/CaptureClient.cpp (hunk 1 above)
Comment (ronaldfw) [reply OrbitCaptureClient/CaptureClient.cpp:39]: I assume  you mean in the commit message/PR description? 

(If yes, fixed. If no, please provide more detail.)

**Pull request merged**